extensions/quickfile/src/balances.tsx

```diff
@@ -0,0 +1,94 @@
+import { ActionPanel, List, Action, getPreferenceValues } from "@raycast/api";
+
+import { bankGetAccounts, bankGetAccountTransactions, BankType } from "./api/bank";
+import { systemGetAccountDetails } from "./api/system";
+import { accountPath, iconPath } from "./api/urls";
+
+import { formatCurrency } from "./utils/formatting";
+import { useAsyncData } from "./utils/hooks";
+import { unique } from "./utils/data";
+import AccountDetail from "./views/AccountDetail";
+
+const bankTypePriority: Record<BankType, number> = {
+  CURRENT: 100,
+  PETTY: 80,
+  BUILDINGSOC: 70,
+  CREDITCARD: 60,
+  LOAN: 50,
+  MERCHANT: 30,
+  RESERVE: 20,
+  EQUITY: 10,
+};
+
+const bankTypeNames: Record<BankType, string> = {
+  CURRENT: "Current accounts",
+  PETTY: "Petty cash",
+  BUILDINGSOC: "Building societies",
+  LOAN: "Loan",
+  MERCHANT: "Merchant account",
+  EQUITY: "Equity",
+  CREDITCARD: "Credit cards",
+  RESERVE: "Reserve",
+};
+
+export default function Command() {
+  const { accountNumber } = getPreferenceValues();
+
+  const { data: domain, loading: loadingDomain } = useAsyncData(accountNumber && "slug", () =>
+    systemGetAccountDetails(accountNumber, ["ClientDomain"]).then((r) => r.ClientDomain)
+  );
+
+  const { data: accounts = [], loading: loadingAccounts } = useAsyncData(accountNumber && "accounts", () =>
+    bankGetAccounts()
+  );
+
+  const bankCodes = accounts.map((a) => a.NominalCode);
+  const { data: transactions = [], loading: loadingTransactions } = useAsyncData(bankCodes, (codes) =>
+    Promise.all(codes.map((code) => bankGetAccountTransactions(code, { number: 20 })))
+  );
+
+  const accountTypes = unique(accounts.map((a) => a.BankType)).sort(
+    (a, b) => (bankTypePriority[b] || 0) - (bankTypePriority[a] || 0)
+  );
+
+  return (
+    <List isLoading={loadingDomain || loadingAccounts || loadingTransactions}>
+      {accountTypes.map((type) => (
```

Comment:
You could add a [List.EmptyView](https://developers.raycast.com/api-reference/user-interface/list#list.emptyview) here to make the UX even better, something like this:

```suggestion
      <List.EmptyView icon={{ source: "icon.png" }} title="No Results" />
      {accountTypes.map((type) => (
```

You could make a copy of the current logo and resize it to 64x64 pixels and it will look something like this:

![CleanShot 2022-05-24 at 05 37 50@2x](https://user-images.githubusercontent.com/38033107/169944049-9ceecb71-62da-4d65-bf38-710db9b0d6d8.png)

If you also make some conditional text, if an error accours then you could change the text to something else 😄